lib/src/model.dart

```diff
@@ -4548,3 +4864,313 @@ class TypeParameter extends ModelElement {
+      this.autoIncludeDependencies);
+
+  void logAnalysisErrors(Set<Source> sources) {}
+
+  Package buildPackage() {
+    Set<LibraryElement> libraries = getLibraries(getFiles);
+    return new Package(
+        libraries, packageMeta, getWarningOptions(), context, sdk);
+  }
+
+  DartSdk _sdk;
+  DartSdk get sdk {
+    if (_sdk == null) {
+      _sdk = new FolderBasedDartSdk(PhysicalResourceProvider.INSTANCE,
+          PhysicalResourceProvider.INSTANCE.getFolder(sdkDir.path));
+    }
+    return _sdk;
+  }
+
+  EmbedderSdk _embedderSdk;
+  EmbedderSdk get embedderSdk {
+    if (_embedderSdk == null && packageMeta.isSdk == false) {
+      _embedderSdk = new EmbedderSdk(PhysicalResourceProvider.INSTANCE,
+          new EmbedderYamlLocator(packageMap).embedderYamls);
+    }
+    return _embedderSdk;
+  }
+
+  static Map<String, List<fileSystem.Folder>> _calculatePackageMap(
+      fileSystem.Folder dir) {
+    Map<String, List<fileSystem.Folder>> map = new Map();
+    var info = package_config.findPackagesFromFile(dir.toUri());
+
+    for (String name in info.packages) {
+      Uri uri = info.asMap()[name];
+      fileSystem.Resource resource =
+          PhysicalResourceProvider.INSTANCE.getResource(uri.toFilePath());
+      if (resource is fileSystem.Folder) {
+        map[name] = [resource];
+      }
+    }
+
+    return map;
+  }
+
+  Map<String, List<fileSystem.Folder>> _packageMap;
+  Map<String, List<fileSystem.Folder>> get packageMap {
+    if (_packageMap == null) {
+      fileSystem.Folder cwd =
+          PhysicalResourceProvider.INSTANCE.getResource(rootDir.path);
+      _packageMap = _calculatePackageMap(cwd);
+    }
+    return _packageMap;
+  }
+
+  DartUriResolver _embedderResolver;
+  DartUriResolver get embedderResolver {
+    if (_embedderResolver == null) {
+      _embedderResolver = new DartUriResolver(embedderSdk);
+    }
+    return _embedderResolver;
+  }
+
+  SourceFactory get sourceFactory {
+    List<UriResolver> resolvers = [];
+    resolvers.add(new SdkExtUriResolver(packageMap));
+    resolvers.add(new PackageMapUriResolver(
+        PhysicalResourceProvider.INSTANCE, packageMap));
+    if (embedderSdk == null || embedderSdk.urlMappings.length == 0) {
+      // The embedder uri resolver has no mappings. Use the default Dart SDK
+      // uri resolver.
+      resolvers.add(new DartUriResolver(sdk));
+    } else {
+      // The embedder uri resolver has mappings, use it instead of the default
+      // Dart SDK uri resolver.
+      resolvers.add(embedderResolver);
+    }
+    resolvers.add(
+        new fileSystem.ResourceUriResolver(PhysicalResourceProvider.INSTANCE));
+
+    SourceFactory sourceFactory = new SourceFactory(resolvers);
+    return sourceFactory;
+  }
+
+  AnalysisContext _context;
+  AnalysisContext get context {
+    if (_context == null) {
+      // TODO(jcollins-g): fix this so it actually obeys analyzer options files.
+      var options = new AnalysisOptionsImpl()..enableAssertInitializer = true;
```

Comment:
@bwilkerson - Janice has changed dartdoc so that it only creates an analysis context in one place now. We should look into unifying with how the analysis server creates contexts - we have several outstanding issues as dartdoc doesn't pick up the analysis options settings from projects.